src/main/java/engine/FormplayerIndexedFixtureInstanceTreeElement.java

```diff
@@ -0,0 +1,51 @@
+package engine;
+
+import org.commcare.cases.instance.IndexedFixtureInstanceTreeElement;
+import org.commcare.cases.model.StorageIndexedTreeElementModel;
+import org.commcare.core.interfaces.UserSandbox;
+import org.javarosa.core.model.IndexedFixtureIdentifier;
+import org.javarosa.core.model.instance.AbstractTreeElement;
+import org.javarosa.core.model.instance.InstanceBase;
+import org.javarosa.core.model.instance.TreeElement;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.util.externalizable.DeserializationException;
+
+import util.SerializationUtil;
+
+public class FormplayerIndexedFixtureInstanceTreeElement extends IndexedFixtureInstanceTreeElement {
+
+    private TreeElement attributes;
+
+    private FormplayerIndexedFixtureInstanceTreeElement(AbstractTreeElement instanceRoot,
+                                                    IStorageUtilityIndexed<StorageIndexedTreeElementModel> storage,
+                                                    IndexedFixtureIdentifier indexedFixtureIdentifier) {
+        super(instanceRoot, storage, indexedFixtureIdentifier);
+    }
+
+    public static IndexedFixtureInstanceTreeElement get(UserSandbox sandbox,
+                                                        String instanceName,
+                                                        InstanceBase instanceBase) {
+        IndexedFixtureIdentifier indexedFixtureIdentifier = sandbox.getIndexedFixtureIdentifier(instanceName);
+        if (indexedFixtureIdentifier == null) {
+            return null;
+        } else {
+            IStorageUtilityIndexed<StorageIndexedTreeElementModel> storage =
+                    sandbox.getIndexedFixtureStorage(instanceName);
+            return new FormplayerIndexedFixtureInstanceTreeElement(instanceBase, storage, indexedFixtureIdentifier);
+        }
+    }
+
+    protected synchronized TreeElement loadAttributes() {
+        if (attributes == null) {
+            try {
+                attributes = SerializationUtil.deserialize(attrHolder, TreeElement.class);
+            } catch (Exception e) {
+                if (e.getCause() instanceof DeserializationException) {
```

Comment:
It seems really scary to me that this swallows errors that aren't deserialization exceptions. What's the reason for that?

Reply:
@ctsims Hey thanks for catching this, that's definitely a mistake. Corrected it.

Reply:
Cool, looks good to me now.